File: archived/ptune-log/src/core/models/tasks/MyTask/PomodoroInfo.ts
export class PomodoroInfo {
  constructor(public planned: number, public actual?: number) {}

  toString(): string {
    const planned = this.planned !== undefined ? `🍅x${this.planned}` : '';
    const actual = this.actual !== undefined ? `✅x${this.actual}` : '';
    return `${planned}${actual ? ' ' + actual : ''}`;
  }

  done(count = 1): void {
    if (this.actual === undefined) this.actual = 0;
    this.actual += count;
  }
}

File: archived/PtuneSync/PtuneSync/Models/PomodoroInfo.cs
using System;
using System.Text.Json.Serialization;

namespace PtuneSync.Models;

/// <summary>
/// Pomodoroの予定回数と実績回数を保持するモデル。
/// </summary>
public class PomodoroInfo
{
    [JsonPropertyName("planned")]
    public int Planned { get; set; }

    [JsonPropertyName("actual")]
    public double? Actual { get; set; }

    public PomodoroInfo(int planned, double? actual = null)
    {
        Planned = planned;
        Actual = actual;
    }

    /// <summary>
    /// 文字列表現を返す (例: "🍅x3 ✅x2")
    /// </summary>
    public override string ToString()
    {
        string plannedPart = Planned > 0 ? $"🍅x{Planned}" : string.Empty;
        string actualPart = Actual.HasValue ? $" ✅x{Actual}" : string.Empty;
        return $"{plannedPart}{actualPart}".Trim();
    }

    /// <summary>
    /// 実績回数を増やす
    /// </summary>
    public void Done(int count = 1)
    {
        Actual ??= 0;
        Actual += count;
    }
}
